report/4.Solution/SemanticsDescription.tex

```diff
@@ -0,0 +1,748 @@
+The scope rules of HCL are fully static.
+This means that the bindings accessed when using variable or functions names, will be the ones that were bound when the variable or function was declared.
+The reader should note that, since HCL is utilizing fully static scope rules, transitions for function calls needs special components to successfully enable recursive calling of function.
+In other words, when make a recursive call in HCL the bindings of the procedure environment will be dynamic.
+
+\subsection{Abstract Syntax}
+The abstract syntax will now be presented.
+For convenience, the following notational conventions will be followed throughout this section.
+
+In the rest of the section $t1, .
+.
+.
+ , tn$ will be abbreviated with $t\textasciitilde$ and $t1 .
+ .
+ .
+  tn$ with $t^*$ ($t^+$ in the non-empty case).
+
+The syntax categories and meta-variables used in the semantic can be seen in table \ref{tab:metaVarab}. 
+\begin{center}
+	\begin{table}[ht]
+		\centering
+		\caption{The syntax categories and meta-variables used in the operational semantics of HCL}
+		\label{tab:metaVar}
+		\begin{tabular}{ll}
+			\textbf{Metavariable} & \textbf{Name} \\
+			$a \in A_{exp}$ 	  & Arithmetic expression \\
+			$n \in Num$			  & Number Literal \\
+			$x \in Var$           & Variable \\
+			$b \in B_{exp}$		  & Boolean expression\\ 
+			$tf \in {True,False}$ & Boolean literal \\
+			$t \in Txt_{exp}$     & Text expression\\
+			$str \in {\Sigma^*}$  & String Literal \\
+			$list \in list_{exp}$ & List expression \\
+			$lst \in List$		  & List literal\\
+			$tuple \in tpl_{exp}$ & Tuple expression\\
+			$tpl \in Tuple$	      & Tuple Literal\\
+			$L \in Lambda$		  & Lambda expression\\
+			$e \in Exp$			  & Expression\\
+			$el \in Lit$		  & Expression Literal\\
+			$p \in Pnames$        & Procedure names\\
+			$L \in Lambda$        & Statement Lambdas\\
+			$S \in Stm$           & Statement\\
+			$f \in Fnames$ 		  & Function names\\
+			$D_V \in Dcl_V$		  & Variable declarations\\
+			$D_P \in Dcl_P$       & Function declarations\\
+			$T \in Type$          & Types
+		\end{tabular}
+	\end{table}
+\end{center}
+
+For the rest of this section, it will be assumed that functions exists with domains equal to the sets of literals associated with the various expressions' syntax categories, and with co-domains equal to the values extracted from those literals.
+The abstract syntax of HCL can be seen below.
+
+$a\ ::=\ n\ |\ x\ |\ a_1+a_2\ |\ a_1-a_2\ |\ a_1*a_2\ |\ a_1/a_2\ |\ (a)$\\
+$b\ ::=\ tf\ |\ x\ |\ a_1<a_2\ |\ a_1>a_2\ |\ a_1<=a_2\ |\ a_1>=a_2\ |\ a_1==a_2\ |\ b_1\ and\ b_2\ |\ b_1\ or\ b_2\ |\ b\ not\ |\ (b)$\\
+$t\ ::=\ str\ |\ x\ |\ t_1+t_2\ |\ t\ toString\ |\ (t)$\\
+$list\ ::=\ lst\ |\ x\ |\ list\ length\ |\ list\ @\ a\ |\ (list)$\\
+$tuple\ ::=\ tpl\ |\ x\ |\ tuple\ @\ a\ |\ (tuple)$\\
+$e\ ::=\ a\ |\ b\ |\ t\ |\ list\ |\ tuple\ |\ F_{exp}\ | L$\\
+$pOrf\ ::=\ f\ |\ p$\\
```

Comment:
maybe explain in a better way that you mean the "P" or the "F" explained in previous table

Reply:
Moved it to syntax catagory table.
Fixed